src/darwin/Framework/CHIP/MTRDevice.h

```diff
@@ -29,8 +29,37 @@ typedef NS_ENUM(NSUInteger, MTRDeviceState) {
     MTRDeviceStateUnreachable = 2
 };
 
+/**
+ * This enum is used to specify the type of logs requested from this device.
+ *
+ * The logs types are : End User Support, Network Diagnostics and Crash logs.
+ */
+typedef NS_ENUM(NSUInteger, MTRDiagnosticLogType) {
+    MTRDiagnosticLogTypeEndUserSupport = 0, // End user support logs are requested
+    MTRDiagnosticLogTypeNetworkDiagnostics = 1, // Network Diagnostics logs are requested
+    MTRDiagnosticLogTypeCrash = 2, // Crash logs are requested
+    MTRDiagnosticLogTypeUnknown = 3
+} MTR_PROVISIONALLY_AVAILABLE;
+
 @protocol MTRDeviceDelegate;
 
+MTR_PROVISIONALLY_AVAILABLE
+@interface MTRDiagnosticLogResult : NSObject
+
+/**
+ * A boolean indicating whether the request for diagnostic logs has timed out.
+ */
+@property (readonly, nonatomic) BOOL requestTimedOut;
```

Comment:
I'm not sure how useful it is to distinguish between timeouts and other errors here... if we download 2 of 3 files and then we get some low-level transport error, wouldn't we still want to return partial results?

So maybe the completion handler could always get just a non-null MTRDiagnosticLogResult (and no separate NSError parameter), and have a nullable `error` property here instead of this boolean. That way we can return partial results along with any kind of error.

Reply:
yes we want to return partial logs with different errors. I guess we could map to a timeout error and return that instead of a boolean.

Reply:
done.